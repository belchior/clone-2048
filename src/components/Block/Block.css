@import '../../static/vars.css';

@media all {
  .Block {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-basis: 25%;
    border: 1px solid var(--gray1);
    color: #ffffff;
  }

  .block-0 { background-color: var(--block-0); color: var(--block-0); }
  .block-2 { background-color: var(--block-2); }
  .block-4 { background-color: var(--block-4); }
  .block-8 { background-color: var(--block-8); }
  .block-16 { background-color: var(--block-16); }
  .block-32 { background-color: var(--block-32); }
  .block-64 { background-color: var(--block-64); }
  .block-128 { background-color: var(--block-128); }
  .block-256 { background-color: var(--block-256); }
  .block-512 { background-color: var(--block-512); }
  .block-1024 { background-color: var(--block-1024); }
  .block-2048 { background-color: var(--block-2048); }
  .block-4096 { background-color: var(--block-4096); }
}
