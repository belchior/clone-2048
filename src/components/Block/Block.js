import React, { Component } from 'react';

import './Block.css';

export class Block extends Component {
  render() {
    return (
      <div className={'Block block-' + this.props.value}>{this.props.value}</div>
    );
  }
}
