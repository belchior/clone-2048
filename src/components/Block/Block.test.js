import React from 'react';
import { Block } from './Block';
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ShallowRenderer();

it('should render Block without crashing', () => {
  let tree = renderer.render(<Block value={0} />);
  expect(tree).toMatchSnapshot();
});

it('should render Block with the value 64', () => {
  let tree = renderer.render(<Block value={65} />);
  expect(tree).toMatchSnapshot();
});
