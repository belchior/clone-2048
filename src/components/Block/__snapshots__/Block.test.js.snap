// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Block with the value 64 1`] = `
<div
  className="Block block-65"
>
  65
</div>
`;

exports[`should render Block without crashing 1`] = `
<div
  className="Block block-0"
/>
`;
